pvlib/iotools/pvgis.py

```diff
@@ -174,3 +179,81 @@ def _parse_pvgis_tmy_basic(src):
         data['time(UTC)'], format='%Y%m%d:%H%M', utc=True)
     data = data.drop('time(UTC)', axis=1)
     return data, None, None, None
+
+
+def read_pvgis_tmy(filename, outputformat='csv'):
+    """
+    Read a file downloaded from PVGIS.
+
+    Parameters
+    ----------
+    filename : str, pathlib.Path, or file-like buffer
+        Name, path, or buffer of file downloaded from PVGIS.
+    outputformat : str, default 'csv'
+        Output format of PVGIS file. Must be in
+        ``['csv', 'basic', 'epw', 'json']``.
+
+    Returns
+    -------
+    data : pandas.DataFrame
+        the weather data
+    months_selected : list
+        TMY year for each month, ``None`` for basic and EPW
+    inputs : dict
+        the inputs, ``None`` for basic and EPW
+    meta : list or dict
+        meta data, ``None`` for basic
+
+    Raises
+    ------
+    ValueError
+        if `outputformat` isn't in ``['csv', 'basic', 'epw', 'json']``
+
+    See also
+    --------
+    get_pvgis_tmy
+    """
+    # parse the pvgis file based on the output format, either 'epw', 'json',
+    # 'csv', or 'basic'
+
+    # EPW: use the EPW parser from the pvlib.iotools epw.py module
+    if outputformat == 'epw':
+        try:
+            data, meta = parse_epw(filename)
+        except AttributeError:  # str/path has no .read() attribute
+            data, meta = read_epw(filename)
+        return data, None, None, meta
+
+    # NOTE: json, csv, and basic output formats have parsers defined as private
+    # functions in this module
+
+    # JSON: use Python built-in json module to convert file contents to a
+    # Python dictionary, and pass the dictionary to the _parse_pvgis_tmy_json()
+    # function from this module
+    if outputformat == 'json':
+        try:
+            src = json.load(filename)
```

Comment:
Is it possible to use `pd.read_json`?

Reply:
I wish, but unfortunately not b/c it's not tabular. At the top level, there's 3 keys: `'inputs'`, `'meta'`, and `'outputs'` and the data we want is in `outputs['tmy_hourly']`. But there's also `months_selected` in outputs too, so it's a mess. I don't know if we really need/want all of this extra data, but pvgis returns it, so that's what I passed on to the end user.